### PR TITLE
chore(core): pooling of security contexts

### PR DESCRIPTION
**Pull request by glasstiger**
- base class for anonymous multi tenant pools, such as SqlCompilerPool
- a bit of generics refactoring in pools
- separate config for the pools

required by https://github.com/questdb/questdb-enterprise/pull/214

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 16 / 22 (72.73%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SecurityContext.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 16 / 22 (72.73%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SecurityContext.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 77 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SecurityContext.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 77 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/SecurityContext.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 76 (86.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 76 (86.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 1]
```diff
@@ -1704,13 +1704,13 @@ private void handleAuthentication() throws PeerIsSlowToWriteException, PeerIsSlo
         try {
             r = authenticator.handleIO();
             if (r == Authenticator.OK) {
-                CharSequence principal = authenticator.getPrincipal();
-                SecurityContext securityContext = securityContextFactory.getInstance(principal, SecurityContextFactory.PGWIRE);
+                SecurityContext securityContext = securityContextFactory.getInstance(authenticator.getPrincipal(), SecurityContextFactory.PGWIRE);
                 try {
                     securityContext.authorizePGWire();
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:1709]: If this line throws there can be leak of `securityContext`

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 76 (86.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 1 above)
Comment (glasstiger) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:1716]: pushed a fix

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 80 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 95 / 107 (88.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|25|27|92.59%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 97 / 109 (88.99%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|27|29|93.10%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 96 / 108 (88.89%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|25|27|92.59%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 97 / 109 (88.99%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/security/ReadOnlySecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/security/AllowAllSecurityContext.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/pool/SqlCompilerPool.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpConnectionContext.java|26|28|92.86%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractAnonymousMultiTenantPool.java|31|32|96.88%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyRequestJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/AbstractMultiTenantPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/TelemetryJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WalWriterPool.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

**Comment by glasstiger** [comment]
superseded by https://github.com/questdb/questdb/pull/3955

**Pull request closed without merging**